libraries/AP_BattMonitor/AP_BattMonitor_Backend.h

```diff
@@ -34,13 +34,30 @@ class AP_BattMonitor_Backend
     // read the latest battery voltage
     virtual void read() = 0;
 
+    /// returns true if battery monitor instance provides current info
+    virtual bool has_current() const = 0;
+
+    // returns true if battery monitor provides individual cell voltages
+    virtual bool has_cell_voltages() const { return false; }
+
     /// capacity_remaining_pct - returns the % battery capacity remaining (0 ~ 100)
     uint8_t capacity_remaining_pct() const;
 
     /// get capacity for this instance
     int32_t get_capacity() const;
 
+    // update battery resistance estimate and voltage_resting_estimate
+    void update_resistance_estimate();
+
 protected:
     AP_BattMonitor                      &_mon;      // reference to front-end
     AP_BattMonitor::BattMonitor_State   &_state;    // reference to this instances state (held in the front-end)
+
+    // resistance estimate
```

Comment:
Not very important but all of below can be made private, preventing them being changed from backends.